chirp/drivers/tdh8.py

```diff
@@ -906,7 +893,47 @@ def get_memory(self, number):
         elif int(scan_val_list[number - 1]) == 1:
             mem.skip = ''
 
-        mem.mode = _mem.wide and "NFM" or "FM"
+        if self._is_txinh(_mem):
+                mem.duplex = ""
+                mem.offset = 0
+        elif int(_mem.rxfreq) == int(_mem.txfreq):
+            mem.duplex = ""
+            mem.offset = 0
+        elif abs(int(_mem.rxfreq) * 10 - int(_mem.txfreq) * 10) > 70000000:
+            mem.duplex = "split"
+            mem.offset = int(_mem.txfreq) * 10
+        else:
+            mem.duplex = int(_mem.rxfreq) > int(_mem.txfreq) and "-" or "+"
+            mem.offset = abs(int(_mem.rxfreq) - int(_mem.txfreq)) * 10
+
+        # Offset
+        if self._gmrs:
+            mem.duplex = ""
+            mem.offset = 0
+            mem.immutable = ['duplex']
+            if mem.number >= 1 and mem.number <= 30:
+                mem.immutable = ['freq']
+                if mem.number >= 8 and mem.number <= 14:
+                    mem.mode = 'NFM'
+                    mem.power = TX_POWER[0]
+                    mem.immutable = ['freq', 'mode', 'power', 'duplex',
+                                     'offset']
+            elif mem.number >= 31 and mem.number <= 53:
+                self.validate_memory(mem)
```

Comment:
This doesn't make any sense. CHIRP calls `validate_memory()` before `set_memory()` to see if it is suitable. It doesn't make any sense to call it in `get_memory()` because *you* are returning the memory, so you control the rules.